### DISCUSSION TITLE
BetterTTV Safari Extension

### DISCUSSION THREAD

**Discussion by Wilkolicious** [night/betterttv#5842] [Ideas]
Safari Extensions can no longer be installed from outside the App Store / Safari Extensions Gallery on Safari 12.0.

> Support for developer-signed .safariextz Safari Extensions in Safari 12 on macOS has been removed. They no longer appear in Safari preferences and cannot be enabled. On first launch users will receive a warning notification and these extension will not load.

See: https://developer.apple.com/safari/whats-new/

The only work-around seems to be through the extension builder.  Details here: https://georgegarside.com/blog/macos/install-any-safari-extension-macos-mojave/

**Comment by night** [comment]
Safari now forces their extension development into their native apps with XCode, which is a strange pivot from how all other browser do extension developer. Additionally, they've introduced a $99/year fee to list your extension in their app store.. which is unheard of across all the browsers. Given the rough number of people who use Safari it doesn't currently make sense for us to spend the time and effort to build and maintain a Safari extension any longer.

**Comment by BNTFryingPan** [comment]
Safari is the Internet Explorer of MacOS, Heck apple even used IE on the earlier Macs before they had Safari, so its not surprising that Safari is a bad web browser, although its better than IE.

Edge is probably the best default Web Browser ive seen on an OS (other than Firefox on most Linux Distros)

**Comment by Juice805** [comment]
FWIW I use it in Safari. It's great. Definitely more usable than Edge. Although I'm not sure what the quality of either software has to do with this "bug".

Could always sell it for $1 on the store and make it back.

**Comment by AdamChandler1986** [comment]
I know Safari is not important. I hope you'll re-consider supporting this in the future. Safari just removed your extension from my browser. technical Mac users use Safari because it has the lowest resource use of any other browser. Chrome & FireFox cut our battery life in half and Chrome is very user-privacy-hostile to a point that privacy-conscious users choose Safari. I wish Apple didn't gate-keep access to its users with a $99 tax which you and I know is completely wrong for them to do. 

I do feel for you though. Here's Reddit Enhancement Suite's Developer on Safari's Changes and their depreciation of support for it - https://redditenhancementsuite.com/safari/

**Comment by dljsjr** [comment]
I'd also like to throw my hat in to the ring for continued Safari support. I know that it's a smaller browser market and the $99 toll bridge is definitely a developer hostile move. But my daily driver is a MacBook and I use Safari for its integration with the rest of my iOS devices and the battery life/memory use concerns.

I'd be more than happy to contribute to a Patreon or something if that was what was needed to keep Safari support alive.

That said as a developer myself I'm sympathetic to the idea that you wouldn't want to support Safari as a matter of principle. I just wanted to chime in and express interest in keeping support alive.

**Comment by Vardek13** [comment]
Hi,

Related to this issue, FrankerfaceZ found a workaround through Tampermonkey, so if you got that extension, safari users would only need the user script in order to use the extension like it was before. I am not very knowledgable on the subject, but is the user script for betterttv accessible already? If not is it possible to make it accessible?

Thanks in advance.

**Comment by MysticalOS** [comment]
Yeah I actually used that work around for some other stuff too. Like socialfixer has a monkey script you can use instead of extension. It works as a work around since Tampermonkey is in the gallery so can be used with safari 12.0

**Comment by samfundev** [comment]
Here is a [userscript](https://gist.github.com/samfundev/8541e9d78f0f9aee5fd4c262b7ebc95e/raw/04ab5d00aa16ffeb843683f78b88af7de9e80333/BetterTTV.user.js) which should do the same thing as the chrome extension. [Source](https://gist.github.com/samfundev/8541e9d78f0f9aee5fd4c262b7ebc95e)

**Comment by Vardek13** [comment]
Thank you that worked perfectly! Honestly, I think this is the better solution for everybody.
